### PR TITLE
Fix FXIOS-8779 - Login autofill not working on certain websites

### LINKED ISSUE #19276: The `Use saved password` button is not displayed under certain steps
## Steps to reproduce
1. Access https://63p8v.csb.app/ and tap on the username or password field.
1. Tap on the Use saved password.
1. Close the `Use saved password` prompt (tap on x).
1. Tap on the username or password field.
### Expected behavior
The `Use saved password` button is displayed. 
### Actual behavior
The `Use saved password` button is not displayed under certain steps.
### Device & build information

* Device: iPhone 15 Pro (17.3) 
* Build version: v124 (39709) 
* First seen version: N/A 
### Notes
Attachments: 

https://github.com/mozilla-mobile/firefox-ios/assets/42831109/5c1bf8e4-b72b-4e3a-a5be-07e92cf40526

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-8690)

### LINKED ISSUE #19448: [Password Autofill] The autofill can be done only on one field at a time on `Netflix.com` even when having a username and password on the same webpage
## Steps to reproduce
1. Launch FF and save a login for netflix.com.
2. Access netflix.com and then sign in.
3. Tap on the username and then on the `Use saved password`.
4. Select the account in order to autofill.
### Expected behavior
Both password and username fields are auto-filled without a problem.
### Actual behavior
The autofill can be done only on one field at a time on `Netflix.com` even when having a username and password on the same webpage.
### Device & build information
* Device: iPhone 15 Pro (17.3). 
* Build version: 125 (40129) 
* First seen version: N/A 
### Notes
Attachments: 

https://github.com/mozilla-mobile/firefox-ios/assets/42831109/0a175606-7d1a-4621-ae67-f166a16220dd

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-8778)

### LINKED ISSUE #19450: [Password Autofill] The autofill is not done on some websites
## Steps to reproduce
1. Launch FF and save an account and password for a website.
1. Tap on the username or password fields.
1. Tap on the `Use saved password`.
1. Select the saved account for the website and attempt to autofill.
### Expected behavior
The autofill is done without problems.
### Actual behavior
 The autofill is not done on some websites.
### Device & build information

* Device: iPhone 15 Pro (17.3). 
* Build version: 125 (40129) 
* First seen version: N/A 
### Notes

|` Website` | `Use saved password button` appear? | `Autofill Done?` |
| :-------------: | :-------------: | :-------------: |
| `Accounts.google.com`  |:white_check_mark:|    :x:  |
| `youtube.com`             |        :white_check_mark:            |        :x:            |
|`baidu.com`                |         :white_check_mark:           |        :x:            |
| `wipipedia.org`               |        :white_check_mark:            |         :x:           |
| `yandex.ru`               |          :white_check_mark:          |        :x:            |
| `tiktok.com`               |         :white_check_mark:           |        :x:            |
| `openai.com`               |        :white_check_mark:            |     :x:               |
| `netflix.com`               |      :white_check_mark:              |        :x:            |
| `office.com`               |        :white_check_mark:            |        :x:            |
| `dzen.ru`               |          :white_check_mark:          |         :x:           |
| `zoom.us`               |        :white_check_mark:            |        :x:            |
| `naver.com`               |      :white_check_mark:              |      :x:              |
| `mail.ru`               |             :white_check_mark:       |         :x:           |
|  `ebay.com`              |         :white_check_mark:           |      :x:              |

Attachments: 

https://github.com/mozilla-mobile/firefox-ios/assets/42831109/668858f9-1d18-4014-b5a6-37d858779ee0

https://github.com/mozilla-mobile/firefox-ios/assets/42831109/39399d1c-f66d-42fc-99d4-e92098c7d070

https://github.com/mozilla-mobile/firefox-ios/assets/42831109/ea78f476-1bc1-4215-b477-6cec883b1730

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-8779)

### PR DESCRIPTION
**Pull request by issammani**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8779)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/19450)

## :bulb: Description
This PR:
- Improves login autofill for react based apps.
- Listens for `focusin` event on capture since the page might cancel it.
- Fixes #19276
- Fixes #19448
- Fixes #19450

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/TabContentsScripts/LoginsHelper.swift [hunk 1]
```diff
@@ -338,6 +338,11 @@ class LoginsHelper: TabContentScript {
                                      object: .loginsAutofilled)
     }
 
+    public static func yieldFocusBackToField(with tab: Tab) {
+        let jsFocusCallback = "window.__firefox__.logins.yieldFocusBackToField()"
+        tab.webView?.evaluateJavascriptInDefaultContentWorld(jsFocusCallback)
+    }
+
```
Comment (issammani) [inline firefox-ios/Client/Frontend/TabContentsScripts/LoginsHelper.swift:345]: This is for [FXIOS-8690](https://mozilla-hub.atlassian.net/browse/FXIOS-8690). When we dismiss the bottom sheet we also want to yield focus back to the field and show the accessory view.

File: firefox-ios/Client/Coordinators/CredentialAutofillCoordinator.swift [hunk 2]
```diff
@@ -155,6 +155,11 @@ class CredentialAutofillCoordinator: BaseCoordinator {
 
         let viewController = SelfSizingHostingController(rootView: loginAutofillView)
 
+        viewController.controllerWillDismiss = { [weak self] in
+            guard let currentTab = self?.tabManager.selectedTab else { return }
+            LoginsHelper.yieldFocusBackToField(with: currentTab)
+        }
```
Comment (nbhasin2) [inline firefox-ios/Client/Coordinators/CredentialAutofillCoordinator.swift:161]: @razvanlitianu to doublecheck

**Review by nbhasin2 — APPROVED** [review]

**Comment by afurlan-firefox** [comment]
Is this one going on 125?

**Comment by nbhasin2** [comment]
@afurlan-firefox yes

@mergify backport release/v125

**Review by razvanlitianu — APPROVED** [review]

**Comment by razvanlitianu** [comment]
@mergify backport release/v125

**Comment by razvanlitianu** [comment]
@afurlan-firefox @nbhasin2 we have a recent issue that autofill does not work for 124.3, this can be a possible fix. Should we uplift?

**Comment by dsmithpadilla** [comment]
@mergify backport release/v124

**Pull request merged**